src/main/java/org/kohsuke/groovy/sandbox/SandboxTransformer.java

```diff
@@ -704,7 +704,7 @@ public void visitExpressionStatement(ExpressionStatement es) {
                     if (mightBePositionalArgumentConstructor((VariableExpression) leftExpression)) {
                         CastExpression ce = new CastExpression(leftExpression.getType(), de.getRightExpression());
                         ce.setCoerce(true);
-                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), transform(ce))));
+                        es.setExpression(transform(new DeclarationExpression(leftExpression, de.getOperation(), ce)));
```

Comment:
I am positive, from debug testing and inspecting the resulting expressions, but yeah, what we *really* need, IMO, both here and in `groovy-cps`, is AST transformation testing, not just testing of the resulting code.